src/lib/containers/markdown/UserMentionModal.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useCallback } from 'react'
+import { Modal } from 'react-bootstrap'
+import { UserGroupHeader } from '../../utils/synapseTypes'
+import UserSearchBoxV2 from '../UserSearchBoxV2'
+
+export type UserMentionModalProps = {
+  show: boolean
+  onClose: () => void
+  handleUserTag: (user: string) => void
+}
+
+export const UserMentionModal: React.FC<UserMentionModalProps> = ({
+  show,
+  onClose,
+  handleUserTag,
+}: UserMentionModalProps) => {
+  const onUserChange = useCallback(
+    (selected: string | null, header: UserGroupHeader | null) => {
+      if (selected && header) {
+        handleUserTag(header.userName)
+      }
+      onClose()
+    },
+    [onClose, handleUserTag],
+  )
+
+  return (
+    <>
+      <Modal show={show} onHide={onClose} backdrop="static" animation={false}>
+        <Modal.Header closeButton>
+          <Modal.Title>Find User or Team</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <UserSearchBoxV2
```

Comment:
Maybe set the TYPE_FILTER to USERS_ONLY until we work out the Team renderer.  My guess is that the markdown processor is working correctly, but we are not filling in the team badge in SRC